### PR TITLE
Migrate to C++17 & WebRTC @ 'branch-heads/5112

### PR DESCRIPTION
**Pull request by regexident**

### CODE REVIEW

File: include/Producer.hpp [hunk 1]
```diff
@@ -41,6 +43,9 @@ namespace mediasoupclient
 		  webrtc::RtpSenderInterface* rtpSender,
 		  webrtc::MediaStreamTrackInterface* track,
 		  const nlohmann::json& rtpParameters,
+		  const bool stopTracks,
+		  const bool zeroRtpOnPause,
+		  const bool disableTrackOnPause,
```
Comment (jpgneves) [inline include/Producer.hpp:48]: Bad rebase? There's quite a bit of previously-merged code here. Can we have it cleaned up before a final review?

Comment (regexident) [reply include/Producer.hpp:48]: Possible. Moving the commits to the other branch was messy, tbh. It wouldn't allow a simple cherry-pick or rebase for some reason (the commits containing a merge commit, iirc).

Comment (regexident) [reply include/Producer.hpp:48]: Better now?

Comment (jpgneves) [reply include/Producer.hpp:48]: Apparently still the same! But it's reviewable overall, I think. :)

**Review by jpgneves — APPROVED** [review]
LGTM!

File: include/Producer.hpp (hunk 1 above)
Comment (regexident) [reply include/Producer.hpp:48]: I just rebased `3.3.0-daily` onto `3.4.0` and added two more commits on top for `5112`.

**Pull request closed without merging**